src/main/java/net/snowflake/client/core/HttpUtil.java

```diff
@@ -265,6 +267,7 @@ private static String buildUserAgent() {
     String languageVersion =
         (systemGetProperty("java.version") != null) ? systemGetProperty("java.version") : "";
     builder.append(languageVersion);
+    builder.append(null == customSuffix ? "" : " " + customSuffix);
```

Comment:
Please check for empty string. That's default. So no need to give extra space for an empty user agent suffix. Also check with Harsh whether changing URL can cause any backward compatibility issue or not.

Reply:
@sfc-gh-hchaturvedi I'm changing the user agent header by allowing appending of a custom suffix. Are there any backward compatibility issues that I should be concerned?

Reply:
@sfc-gh-igarish why do you think there's a change in the URL? What I see here is a change in how we populate a HTTP header, which shouldn't alter the request.